### PR TITLE
move FS modules into settings

### PR DESCRIPTION
**Pull request by CWelshE**

### CODE REVIEW

File: src/library_fs.js [hunk 1]
```diff
@@ -4,14 +4,8 @@ mergeInto(LibraryManager.library, {
...
-                // export some names through closure
-                'Module["FS_createFolder"] = FS.createFolder;' +
-                'Module["FS_createPath"] = FS.createPath;' +
-                'Module["FS_createDataFile"] = FS.createDataFile;' +
-                'Module["FS_createPreloadedFile"] = FS.createPreloadedFile;' +
-                'Module["FS_createLazyFile"] = FS.createLazyFile;' +
-                'Module["FS_createLink"] = FS.createLink;' +
-                'Module["FS_createDevice"] = FS.createDevice;',
+                //Grab module names
+                {{{FS_MODULES.map(function(module){return 'Module["FS_' + module + '"] = FS.' + module + ";"}).reduce(function(str, module){return str + module;})}}},
```
Comment (kripken) [inline src/library_fs.js]: space after `{{{` and before `}}}`

**Comment by kripken** [comment]
How about calling it `EXPORTED_FS_METHODS`?

**Comment by CWelshE** [comment]
Sounds good to me! Will update PR in a bit

File: src/library_fs.js [hunk 2]
```diff
@@ -4,14 +4,8 @@ mergeInto(LibraryManager.library, {
...
                 '__ATEXIT__.push(function() { FS.quit() });' +
-                // export some names through closure
-                'Module["FS_createFolder"] = FS.createFolder;' +
-                'Module["FS_createPath"] = FS.createPath;' +
-                'Module["FS_createDataFile"] = FS.createDataFile;' +
-                'Module["FS_createPreloadedFile"] = FS.createPreloadedFile;' +
-                'Module["FS_createLazyFile"] = FS.createLazyFile;' +
-                'Module["FS_createLink"] = FS.createLink;' +
-                'Module["FS_createDevice"] = FS.createDevice;',
+                //Grab module names
```
Comment (kripken) [inline src/library_fs.js]: comment and text need to be updated to the new name

**Comment by kripken** [comment]
Please also squash the commits and add yourself to AUTHORS.

**Comment by CWelshE** [comment]
(Hopefully) done! Sorry, this is my first contribution :)

**Comment by kripken** [comment]
I tested this locally and it seems to break emcc (`emcc tests/hello_world.c` fails, for example). I think the issue is that the output of `{{{ x }}}` is the eval of `x`. But here we need a string, so you probably want `'{{{ x }}}'` or `"{{{ x }}} "`.

**Comment by CWelshE** [comment]
Would that be interpreted, despite being inside of a string? (I've been trying to build from source but have been unsuccessful, so I can't test yet)

**Comment by kripken** [comment]
Thanks! Merged to incoming.

**Pull request closed without merging**